recipes/09/09_group_by_session_window.md

```diff
@@ -0,0 +1,57 @@
+# 09 Aggregating Time Series Data (Session Windows)
+
+:bulb: This example will show how to aggregate time-series data in real-time using a `SESSION` window.
+
+The source table (`server_logs`) is backed by the [`faker` connector](https://github.com/knaufk/flink-faker), which continuously generates rows in memory based on Java Faker expressions.
+
+#### What are Session Windows?
+
+In a [previous recipe](../06/06_group_by_window.md), you learned about _tumbling windows_. Another way to group time-series data is using [_session windows_](https://ci.apache.org/projects/flink/flink-docs-stable/dev/table/sql/queries.html#group-windows), which aggregate records into _sessions_ that represent periods of activity followed by gaps of idleness. Think, for example, of user sessions on a website: a user will be active for a given period of time, then leave the website; and each user will be active at different times. To analyze user behaviour, it's useful to aggregate their actions on the website for each period of activity (i.e. _session_).
+
+Unlike tumbling windows, session windows don't have a fixed duration and are tracked independenlty across keys (i.e. windows of different keys will have different durations).
+
+#### Using Session Windows
+
+To count the number of "Forbidden" (403) requests per user over the duration of a session, you can use the `SESSION` built-in group window function. In this example, a session is bounded by a gap of idleness of 10 seconds (`INTERVAL '10' SECOND`). This means that requests that occur within 10 seconds of the last seen request for each user will be merged into the same session window; and any request that occurs outside of this gap will trigger the creation of a new session window.
+
+> Tip: You can use the `SESSION_START` and `SESSION_END` [auxiliary functions](https://ci.apache.org/projects/flink/flink-docs-stable/dev/table/sql/queries.html#selecting-group-window-start-and-end-timestamps) to check the lower and upper bounds of session windows.
+
+You can use session windows in combination with [watermarks](../07/07_watermarks.md) to indicate how out-of-order you expect the data to be, so that Flink can associate records with the correct session even if these are late.
```

Comment:
```suggestion
```

Reply:
In the spirit of keeping things brief and example-driven, I think, we don't need this sentence.